homeassistant/components/mealie/coordinator.py

```diff
@@ -139,3 +141,26 @@ async def _async_update_data(
         except MealieConnectionError as error:
             raise UpdateFailed(error) from error
         return shopping_list_items
+
+
+class MealieStatisticsCoordinator(MealieDataUpdateCoordinator[Statistics]):
+    """Class to manage fetching Mealie Statistics data."""
+
+    def __init__(self, hass: HomeAssistant, client: MealieClient) -> None:
+        """Initialize coordinator."""
+        super().__init__(
+            hass,
+            name="MealieStatistics",
+            client=client,
+            update_interval=timedelta(minutes=15),
+        )
+
+    async def _async_update_data(
+        self,
+    ) -> Statistics:
+        try:
+            return await self.client.get_statistics()
+        except MealieAuthenticationError as error:
+            raise ConfigEntryAuthFailed from error
+        except MealieConnectionError as error:
+            raise UpdateFailed(error) from error
```

Comment:
Missing coverage here.

Reply:
I want to refactor this a bit in a follow up to move the try except block to the base coordinator. I only noticed yesterday that we had that duplicated